discovery/azure/azure_test.go

```diff
@@ -506,3 +687,192 @@ func (m *mockAzureClient) getVMScaleSetVMNetworkInterfaceByID(ctx context.Contex
 	}
 	return m.networkInterface, nil
 }
+
+func createMockAzureClient(vmResp []armcompute.VirtualMachinesClientListAllResponse, vmssResp []armcompute.VirtualMachineScaleSetsClientListAllResponse, vmssvmResp []armcompute.VirtualMachineScaleSetVMsClientListResponse, logger log.Logger) (client, error) {
+	c := mockAzureClient{}
+
+	mockVMServer := defaultMockVMServer(vmResp)
+	mockVMSSServer := defaultMockVMSSServer(vmssResp)
+	mockVMScaleSetVMServer := defaultMockVMSSVMServer(vmssvmResp)
+
+	vmClient, err := armcompute.NewVirtualMachinesClient("fake-subscription-id", &azfake.TokenCredential{}, &arm.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport: fake.NewVirtualMachinesServerTransport(&mockVMServer),
```

Comment:
(suggestion) I would inline the default... functions since they are called only once.

Reply:
True, but I feel like it makes the code more digestible. I lean more towards leaving the functions as is. WDYT?